main.js

```diff
@@ -1,100 +1,233 @@
+    if (dice.shouldRoll[die] && dice.positionValues[0] !== 0) {
+      dicePositions[die].classList.toggle("select");
 
-// --- Select --- \\
-
-// Add event listener on each die.
-for (let i = 0; i < dicePositions.length; i++) {
-        dicePositions[i].addEventListener("click", () => {
-            // toggle Select as long as die is not blank or have "lock" as a class.
-            if (dice.positionValues[0] === 0) {
-                return
-            } else if (dice.rollBoolean[i]) {
-                dicePositions[i].classList.toggle("select");
-            }
-    });
+      points.selectedTotal = 0;
+      points.calculate();
+      uncommitedPoints.innerHTML = points.turnTotal + points.selectedTotal;
+    }
+  });
 }
 
-// --- Hold --- \\
+// --- HOLD --- \\
 
 // Hold Dice - Switch player
-document.getElementById("hold-button").addEventListener('click',() => {
-    const activePlayer_1 = document.getElementById("player1")
-    const activePlayer_2 = document.getElementById("player2")
-
-    //switch Player
-    activePlayer_1.classList.toggle("active")
-    activePlayer_2.classList.toggle("active")
-
-    //reset on roundTotal and selectedTotal points
-    points.roundTotal = 0
-    points.selectedTotal = 0
-
-})
+document.getElementById("hold-button").addEventListener("click", () => {
+  if (activePlayer1.classList[1] === "active") {
+    scorePlayer1.innerHTML = points.turnTotal + points.selectedTotal;
+  } else {
+    scorePlayer2.innerHTML = points.turnTotal + points.selectedTotal;
+  }
+
+  activePlayer1.classList.toggle("active");
+  activePlayer2.classList.toggle("active");
+
+  points.turnTotal = 0;
+  points.selectedTotal = 0;
+  uncommitedPoints.innerHTML = 0;
+
+  dice.removeAllSelect();
+  dice.positionValues = [0, 0, 0, 0, 0, 0];
+  dice.shouldRoll = [true, true, true, true, true, true];
+  dice.display();
+});
 
-// --- Score --- \\
+// --- SCORE --- \\
 
-// Points information
 points = {
-    player1: 0,
-    player2: 0,
-    roundTotal: 0,
-    selectedTotal: 0
-}
+  player1: 0,
+  player2: 0,
+  turnTotal: 0,
+  selectedTotal: 0,
+
+  calculate: function () {
+    diceSelected = this.diceSelected();
+
+    freqObj = this.freq(diceSelected);
+    console.log(freqObj);
+
+    this.scoring(freqObj);
+  },
+
+  diceSelected: function () {
+    diceSelected = [];
+    for (const die of dice.positions) {
+      if (
+        dicePositions[die].classList[1] === "select" &&
+        dice.shouldRoll[die]
+      ) {
+        diceSelected.push(dice.positionValues[die]);
+      }
+    }
+    return diceSelected;
+  },
 
+  freq: function (diceSelected) {
+    let freqObj = {};
+
+    diceSelected.forEach((die) => {
+      if (freqObj[die]) freqObj[die]++;
+      else freqObj[die] = 1;
+    });
+    return freqObj;
+  },
+
+  scoring: function (freqObj) {
+    freqKeys = Object.keys(freqObj);
+
+    switch (true) {
+      // 6 of a kind
+      case freqObj[`${freqKeys[0]}`] === 6:
```

Comment:
I'd like to see `${freqKeys[0]}` assigned to a variable. Idk what the value is supposed to be.

Reply:
Also, why is this not in the loop? What's special about the `0`th case?